app/assets/stylesheets/govuk-component/_search.scss

```diff
@@ -0,0 +1,184 @@
+.govuk-search {
+  position: relative;
+  margin-bottom: 30px;
+
+  $input-size: 40px;
+
+  .search-label {
+    display: block;
+  }
+
+  .search-input {
+    padding: 6px;
+    margin: 0.5em 0;
+    width: 100%;
+    height: $input-size;
+    border: 0;
+    @include box-sizing(border-box);
+    @include core-19($line-height: (28 / 19), $line-height-640: (28 / 16));
+    background: $white;
+    border-radius: 0; //otherwise iphones apply an automatic border radius
+    @include appearance(none);
+  }
+
+  .search-submit {
+    padding: 6px;
+    border: 0;
+    cursor: pointer;
+  }
+
+  // IE6 + IE7 always get the simplest version, regardless of whether javascript is enabled
+  @if ($is-ie == false) or ($ie-version >= 8) {
+    .js-enabled & {
+      .search-label {
+        position: absolute;
+        left: 15px;
+        top: 1px;
+        z-index: 1;
+        @include core-19($line-height: $input-size, $line-height-640: $input-size);
+        color: $secondary-text-colour;
+      }
+    }
+
+    .search-wrapper {
+      display: table;
+      width: 100%;
+      background: $white;
+    }
+
+    //general class applied to search input and button wrapper
+    .search-element {
+      position: relative;
+      display: table-cell;
+      vertical-align: top;
+    }
+
+    .search-input {
+      margin: 0;
+
+      // the .focus class is added by JS and ensures that the input remains above the label once clicked/filled in
```

Comment:
Minor: I think `hide-search-label` could be a better class name. Or `hide-placeholder-label`.
It's not strictly always on focus.